MechJeb2/MechJebModuleStagingController.cs

```diff
@@ -94,23 +94,27 @@ public override void OnUpdate()
             if (HasStayingChutes(StageManager.CurrentStage - 1, vessel))
                 return;
 
-            //only fire decouplers to drop deactivated engines or tanks or fairings
-            bool firesDecoupler = InverseStageFiresDecoupler(StageManager.CurrentStage - 1, vessel);
-            if (firesDecoupler && !(InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel) || HasFairing(StageManager.CurrentStage - 1, vessel)))
-                return;
-
-            //only decouple fairings if the dynamic pressure and altitude conditions are respected
-            if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
-                HasFairing(StageManager.CurrentStage - 1, vessel))
-                return;
+            //Always drop deactivated engines or tanks
+            if (!InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel))
+            {
+                //only decouple fairings if the dynamic pressure and altitude conditions are respected
+                if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
+                    HasFairing(StageManager.CurrentStage - 1, vessel))
+                    return;
+
+                //only release launch clamps if we're at nearly full thrust
+                if (vesselState.thrustCurrent / vesselState.thrustAvailable < 1/1.2 &&

```

Comment:
that likely isn't enough.  1/1.2 is 88% and 88% of a 1.05 SLT is 0.87, which will drop the rocket into the pad.  there should probably be a check here that SLT is > 1.00 at least, and likely what RO users want is to fully spool up while still attached to clamps to maximize thrust while still having the pumps active.  if you drop the clamps early then you waste dV on not-quite-fully-spooled-up engines, and some of the bigger/later engines spool up pretty slowly.   thats why i was thinking 98-99% -- although i'd also almost think it should be a tweakable in some menu someplace.